File: Validation.py
from ExceptionManager import ExceptionManager
from RiceCooker import RiceCooker


class Validation:
    @staticmethod
    def is_cooking(rice_cooker):
        if rice_cooker.cook is not None:
            if rice_cooker.status == RiceCooker.Status.COOKING:
                print("Rice cooker still cooking ...")
                return True
        return False


def get_valid_number_input(prompt_msg, min_val=float("-inf"), max_val=float("inf")):
    while True:
        try:
            user_input = float(input(prompt_msg))
            if min_val <= user_input <= max_val:
                return user_input
            else:
                print("400: Input must be between", min_val, "and", max_val)
        except ValueError:
            print(ExceptionManager.error400_msg_number_input())

File: src/Validation.js
const InputUtils = require("./InputUtils");
const RiceCooker = require("./RiceCooker");
const prompt = require("prompt-sync")({ sigint: true });

class Validation {
  static isCooking(riceCooker) {
    if (
      riceCooker.cook !== null &&
      riceCooker.status === RiceCooker.Status.COOKING
    ) {
      console.log("Rice cooker still cooking ...");
      return true;
    }
    return false;
  }

  static isValidNumber(input) {
    const floatRegex = /^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$/;
    return floatRegex.test(input);
  }

  static getValidNumberInput() {
    console.log(InputUtils.menuPromptMsg());
    let value = prompt();
    while (!this.isValidNumber(value)) {
      console.log("400: Invalid input. Please enter a valid number.");
      value = prompt();
    }
    return parseFloat(value);
  }

  static getValidNumberInputRange(min, max) {
    let input;
    console.log(InputUtils.menuPromptMsgRange(min, max));
    do {
      input = Validation.getValidNumberInput();

      if (input < min || input > max) {
        console.log(`400: Input must be between ${min} and ${max}.`);
      }
    } while (input < min || input > max);

    return input;
  }
}

module.exports = Validation;
